main.cc

```diff
@@ -1894,8 +1894,10 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 view_backlog_broker.stop().get();
             });
 
-            startlog.info("Waiting for gossip to settle before accepting client requests...");
-            gossiper.local().wait_for_gossip_to_settle().get();
+            if (!cfg->check_experimental(db::experimental_features_t::feature::CONSISTENT_TOPOLOGY_CHANGES)) {
+                startlog.info("Waiting for gossip to settle before accepting client requests...");
+                gossiper.local().wait_for_gossip_to_settle().get();
+            }
```

Comment:
Is it enough to check for the experimental feature? Or do we need to also check if the cluster was upgraded?

Reply:
We need to check also other cases like the cluster is not upgraded yet, the error injection `force_gossip_based_join` is enabled.

Changed it to check `storage_service::raft_topology_change_enabled`.